libdevice/imf/_imf_include.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef __IMF_INCLUDE_HPP__
+#define __IMF_INCLUDE_HPP__
+#include "../device_imf.hpp"
+
+#ifdef __LIBDEVICE_IMF_ENABLED__
+
+typedef int16_t VINT16;
+typedef int16_t VSINT16;
+typedef uint16_t VUINT16;
+typedef int32_t VINT32;
+typedef int32_t VSINT32;
+typedef uint32_t VUINT32;
+typedef int64_t VINT64;
+typedef int64_t VSINT64;
+typedef uint64_t VUINT64;
+typedef unsigned char _iml_uint8_t;
+typedef unsigned short _iml_uint16_t;
+typedef unsigned int _iml_uint32_t;
+typedef unsigned long long _iml_uint64_t;
+typedef char _iml_int8_t;
+typedef short _iml_int16_t;
+typedef int _iml_int32_t;
+typedef long long _iml_int64_t;
+typedef union {
+  _iml_uint32_t w;
+  float f;
+} int_float;
+typedef union {
+  _iml_uint64_t w;
+  _iml_uint32_t w32[2];
+  _iml_int32_t s32[2];
+  double f;
+} int_double;
+typedef struct tag_iml_spbits_t {
+  _iml_uint32_t significand : 23;
+  _iml_uint32_t exponent : 8;
+  _iml_uint32_t sign : 1;
+} _iml_spbits_t;
+typedef struct tag_iml_dpbits_t {
+  _iml_uint32_t lo_significand : 32;
+  _iml_uint32_t hi_significand : 20;
+  _iml_uint32_t exponent : 11;
+  _iml_uint32_t sign : 1;
+} _iml_dpbits_t;
+typedef struct tag_iml_ldblbits_t {
+  _iml_uint32_t lo_significand;
+  _iml_uint32_t hi_significand;
+  _iml_uint32_t exponent : 15;
+  _iml_uint32_t sign : 1;
+  _iml_uint32_t pad : 16;
+} _iml_ldblbits_t;
+typedef _iml_spbits_t iml_fp32;
+typedef _iml_dpbits_t iml_fp64;
+typedef _iml_ldblbits_t iml_fp80;
+typedef struct tag_iml_dpdwords_t {
+  _iml_uint32_t lo_dword;
+  _iml_uint32_t hi_dword;
+} _iml_dpdwords_t;
+typedef union {
+  _iml_uint32_t hex[2];
+  _iml_dpbits_t bits;
+  _iml_dpdwords_t dwords;
+  double fp;
+} _iml_dp_union_t;
+typedef union {
+  _iml_uint32_t hex[1];
+  _iml_spbits_t bits;
+  float fp;
+} _iml_sp_union_t;
+inline float as_float(uint32_t a) { return *reinterpret_cast<float *>(&a); }
```

Comment:
these functions should be replaced by corresponding `__builtin_bit_cast`

Reply:
Hi, @zettai-reido  and @akolesov-intel 
The __builtin_bit_cast does the same thing as original "reinterpret_cast" code in IR level, if you agree to replace all reinterpret_cast with __builtin_bit_cast for all "as_XXX" functions, I will update the patch.

Thanks very much.

Reply:
We did it internally and observed no problems after replacement. Let's do that in LLVM also please. Thanks!

Reply:
Hi @jinge90,
we agreed to replace them.
@akolesov-intel  also tested the latest IMF with them and no issues found.
Thank you!

Reply:
All as_XX functions are using __builtin_bit_cast now.